packages/flame/lib/src/extensions/vector2.dart

```diff
@@ -80,6 +80,26 @@ extension Vector2Extension on Vector2 {
     }
   }
 
+  /// Project this onto [other].
+  ///
+  /// If [out] is specified, it will be used to provide the result.
+  Vector2 projection(Vector2 other, {Vector2? out}) {
+    final dotProduct = dot(other);
+    final result = (out?..setFrom(other)) ?? other.clone();
+    return result..scale(dotProduct / other.length2);
+  }
```

Comment:
What if `other` is zero? If we don't want to allow this case, then at least mention it in the docs.

Reply:
Added an assert and to the docs.